src/main/java/cinema/controller/MovieSessionController.java

```diff
@@ -0,0 +1,59 @@
+package cinema.controller;
+
+import cinema.model.MovieSession;
+import cinema.model.dto.request.MovieSessionRequestDto;
+import cinema.model.dto.response.MovieSessionResponseDto;
+import cinema.service.MovieSessionService;
+import cinema.service.mapper.MovieSessionMapper;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    public void create(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = movieSessionMapper.toMovieSessionEntity(movieSessionRequestDto);
+        movieSessionService.add(movieSession);
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAll(@RequestParam Long movieId,
+                                                @RequestParam @DateTimeFormat(pattern = "dd.MM."
+                                                        + "yyyy") LocalDate date) {
+        List<MovieSession> allAvailableSessions = movieSessionService
+                .findAvailableSessions(movieId, date);
+        return allAvailableSessions.stream()
+                .map(movieSessionMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping
+    public void update(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.update(movieSessionMapper.toMovieSessionEntity(movieSessionRequestDto));
+    }
+
+    @DeleteMapping
+    public void delete(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.delete(movieSessionMapper.toMovieSessionEntity(movieSessionRequestDto));
+    }
```

Comment:
Check [this](https://mateacademy.slack.com/archives/G01EHB8CWAE/p1613330083125300) please

Reply:
Fixed